src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx

```diff
@@ -295,15 +279,33 @@ function ComposerWithSuggestions(
         lastTextRef.current = value;
     }, [value]);
 
+    useEffect(() => {
+        const unsubscribe = onyxSubscribe({
+            key: ONYXKEYS.MODAL,
+            callback: (modalArg) => {
+                if (!modalArg) {
+                    return;
+                }
+                isVisibleRef.current = !!modalArg.isVisible;
+                willAlertModalBecomeVisibleRef.current = !!modalArg.willAlertModalBecomeVisible;
```

Comment:
Why don't we use the pattern of `useState` as used [here](https://github.com/Expensify/App/blob/987ff1b536f2144e5fd492c6d2d2aa621d68442f/src/components/Tooltip/EducationalTooltip/BaseEducationalTooltip.tsx#L21C1-L24C8)? Using `useRef` would not help in re-render which we may want to do with the change in `modal`.

Reply:
@rojiphil I checked this pattern but not sure why the perf-test still fails when I use this.

Reply:
> I checked this pattern but not sure why the perf-test still fails when I use this.

@nkdengineer Can you please merge main again and check? A recent PR was merged about 6 hours ago to update the `npm` and `node` version which could be causing this.

Reply:
>  A recent PR was merged about 6 hours ago to update the npm and node version which could be causing this.

@rojiphil It happened when we started the PR.

 I merged the latest main.

Reply:
> It happened when we started the PR.

@nkdengineer Yeah. That's right. Looks like the rendering is happening quite often and we need to avoid unnecessary renders. 

Do you think the following can help here?
1. Using the `loading` status in `useOnyx` as referenced [here](https://github.com/Expensify/react-native-onyx?tab=readme-ov-file#subscribing-to-data-changes) to avoid unnecessary rendering until data is fetched? 
2. Selectively using `isVisible` and `willAlertModalBecomeVisible` as per the pattern [here](https://github.com/Expensify/App/blob/987ff1b536f2144e5fd492c6d2d2aa621d68442f/src/components/Tooltip/EducationalTooltip/BaseEducationalTooltip.tsx#L21C1-L24C8) in `modal` to avoid other unnecessary properties?

![Screenshot 2024-10-22 at 5 19 19 PM](https://github.com/user-attachments/assets/1261d3d8-2df3-478e-882a-f74355471454)

Reply:
@rojiphil I tried these ways but the perf-test isn't fixed. Can we accept this and merge the PR? cc @lakchote

Reply:
Ideally, we would want the fix for the performance issue. But if we want to merge, we may not want to merge with the performance issue. Instead, it may be better to revert the migration changes of `withOnyx` to `useOnyx`. I think this will help resolve the `performance` issue although this will bring up the `lint` issue. But this may be acceptable as we would have fixed the issue impacting the CRITICAL flow as mentioned [here](https://github.com/Expensify/App/issues/50346#issuecomment-2400502842). 
I will let @lakchote weigh in here and advice on the way ahead.

Reply:
We for sure want to fix the performance issue. 

>  Instead, it may be better to revert the migration changes of withOnyx to useOnyx. 

Let's do that if that fixes the issue.

Reply:
Thanks for the details @rojiphil. I misinterpreted what you were saying about `revert the migration changes of withOnyx to useOnyx.`.

Let's do it! @nkdengineer if you could do it in a timely manner, it'd be greatly appreciated.

Reply:
> I misinterpreted what you were saying about `revert the migration changes of withOnyx to useOnyx.`.

Oh! Got it. Wrong sentence formation. My bad